### PR TITLE
fix: StateWaitMsg RPC API

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

As part of https://github.com/ChainSafe/forest/issues/3639

Note: `return_dec` field is ignored in `api compare` tests, see https://github.com/ChainSafe/forest/pull/3801#issuecomment-1846905036

Changes introduced in this pull request:

- Fix `Filecoin.StateWaitMsg` RPC API
- Restore `api compare` tests
- Fix `Receipt` deserialization logic

```console
     Running `target/debug/forest-tool api compare /home/me/fr/snapshots/calibnet/forest_snapshot_calibnet_2023-12-08_height_1156570.forest.car.zst --filter StateWait -n 200`
| RPC Method                  | Forest | Lotus |
|-----------------------------|--------|-------|
| Filecoin.StateWaitMsg (469) | Valid  | Valid |
```

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by lemmih** [comment]
Hmm, we should wait with merging this PR until #3784 is addressed.

**Review by lemmih — APPROVED** [review]

File: src/chain/store/chain_store.rs [hunk 1]
```diff
@@ -485,17 +485,25 @@ where
...
-pub fn get_parent_reciept<DB>(
+pub fn get_parent_receipt<DB>(
     db: &DB,
     block_header: &BlockHeader,
     i: usize,
 ) -> Result<Option<Receipt>, Error>
 where
     DB: Blockstore,
 {
+    // Try Receipt_v2 first
```
Comment (LesnyRumcajs) [inline src/chain/store/chain_store.rs:496]: I'm not sure the logic for containing this should be in the `chain_store`. Seems more like a `shim` thingy.

File: src/message/chain_message.rs [hunk 2]
```diff
@@ -32,6 +32,10 @@ impl ChainMessage {
             ChainMessage::Signed(msg) => msg.cid(),
         }
     }
+
+    pub fn equal_call(&self, other: &Self) -> bool {
```
Comment (LesnyRumcajs) [inline src/message/chain_message.rs:36]: Perhaps some docs? I have no idea what it does.

**Review by LesnyRumcajs — APPROVED** [review]

File: src/chain/store/chain_store.rs (hunk 1 above)
Comment (hanabi1224) [reply src/chain/store/chain_store.rs]: Fixed

File: src/message/chain_message.rs (hunk 2 above)
Comment (hanabi1224) [reply src/message/chain_message.rs:41]: Fixed

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by hanabi1224** [comment]
@LesnyRumcajs Resolved a merge conflict in `Cargo.lock`, please approve again, thanks!

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**